app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageInputRow.kt

```diff
@@ -118,32 +124,55 @@ fun MessageComposerInputRow(
                 onSelectedLineIndexChanged = onSelectedLineIndexChanged,
                 onLineBottomYCoordinateChanged = onLineBottomYCoordinateChanged
             )
-
-            transition.AnimatedVisibility(
-                visible = { it.isEditMessage },
-                enter = fadeIn(),
-                exit = fadeOut()
-            ) {
-                MessageEditActions(
-                    onEditSaveButtonClicked = onEditSaveButtonClicked,
-                    onEditCancelButtonClicked = onEditCancelButtonClicked,
-                    editButtonEnabled = messageComposeInputState.editSaveButtonEnabled
-                )
+            AnimatedVisibility(messageComposeInputState.isEditMessage) {
+                EditExtraOptions(messageComposeInputState.editSaveButtonEnabled, onEditSaveButtonClicked, onEditCancelButtonClicked)
             }
+
         }
-        transition.AnimatedVisibility(
-            visible = { it.isNewMessage },
-            enter = fadeIn(),
-            exit = fadeOut()
-        ) {
-            MessageSendActions(
-                onSendButtonClicked = onSendButtonClicked,
-                sendButtonEnabled = messageComposeInputState.sendButtonEnabled
-            )
+        if (messageComposeInputState is MessageComposeInputState.Active) {
+            when (val type = messageComposeInputState.type) {
+                is MessageComposeInputType.NewMessage ->
+                    MessageSendActions(
+                        onSendButtonClicked = onSendButtonClicked,
+                        sendButtonEnabled = messageComposeInputState.sendButtonEnabled
+                    )
+
+                is MessageComposeInputType.SelfDeletingMessage -> {
+                    Row(verticalAlignment = Alignment.CenterVertically) {
+                        Text(
+                            text = type.selfDeletionDuration.label,
+                            style = typography().label02,
+                            color = colorsScheme().primary,
+                            modifier = Modifier
+                                .padding(horizontal = 16.dp)
```

Comment:
Use given spacing dimensions, otherwise Detekt will probably complain about the magic numbers 😅